### PR TITLE
Deprecate #bootstrap in favour of methods exposed on client/server level

### PR DESCRIPTION
**Pull request by violetagg**
- options, attributes can be added with `#attr`, `#selectorAttr`, `#option` and `#selectorOption`
- local address can be added with `#bindAddress`
- group can be added with `#runOn`
- remoteAddress can be added with `#remoteAddress`
- resolver can be added with `#resolver`

### CODE REVIEW

**Comment by violetagg** [comment]
@simonbasle @rstoyanchev @mp911de PTAL

File: src/main/java/reactor/netty/tcp/TcpClient.java [hunk 1]
```diff
@@ -165,7 +178,9 @@ public final TcpClient remoteAddress(Supplier<? extends SocketAddress> remoteAdd
 	 * enriched bootstrap.
 	 *
 	 * @return a new {@link TcpClient}
+	 * @deprecated  as of 0.9.10. Use the methods exposed on {@link TcpClient} level.
```
Comment (simonbasle) [inline src/main/java/reactor/netty/tcp/TcpClient.java:181]: should probably have a note along the lines of "will be removed in 1.1.0"?

File: src/main/java/reactor/netty/tcp/TcpServer.java [hunk 2]
```diff
@@ -141,7 +141,9 @@ public final TcpServer bindAddress(Supplier<? extends SocketAddress> bindAddress
 	 * enriched bootstrap.
 	 *
 	 * @return a new {@link TcpServer}
+	 * @deprecated  as of 0.9.10. Use the methods exposed on {@link TcpServer} level.
```
Comment (simonbasle) [inline src/main/java/reactor/netty/tcp/TcpServer.java:144]: same as above, comment on which version to expect a removal

File: src/main/java/reactor/netty/udp/UdpClient.java [hunk 3]
```diff
@@ -136,7 +149,9 @@ public final UdpClient remoteAddress(Supplier<? extends SocketAddress> remoteAdd
 	 * enriched bootstrap.
 	 *
 	 * @return a new {@link UdpClient}
+	 * @deprecated  as of 0.9.10. Use the methods exposed on {@link UdpClient} level.
```
Comment (simonbasle) [inline src/main/java/reactor/netty/udp/UdpClient.java:152]: same

File: src/main/java/reactor/netty/udp/UdpServer.java [hunk 4]
```diff
@@ -136,7 +136,9 @@ public final UdpServer bindAddress(Supplier<? extends SocketAddress> bindAddress
 	 * enriched bootstrap.
 	 *
 	 * @return a new {@link UdpServer}
+	 * @deprecated  as of 0.9.10. Use the methods exposed on {@link UdpServer} level.
```
Comment (simonbasle) [inline src/main/java/reactor/netty/udp/UdpServer.java:139]: same

File: src/main/java/reactor/netty/tcp/TcpClient.java (hunk 1 above)
Comment (violetagg) [reply src/main/java/reactor/netty/tcp/TcpClient.java:181]: This method is kept in master for R2DBC, so I'll add 1.1.0

File: src/main/java/reactor/netty/tcp/TcpServer.java (hunk 2 above)
Comment (violetagg) [reply src/main/java/reactor/netty/tcp/TcpServer.java:144]: This method is removed in master, so I'll add 1.0.0

File: src/main/java/reactor/netty/udp/UdpClient.java (hunk 3 above)
Comment (violetagg) [reply src/main/java/reactor/netty/udp/UdpClient.java:152]: This method is removed in master, so I'll add 1.0.0

File: src/main/java/reactor/netty/udp/UdpServer.java (hunk 4 above)
Comment (violetagg) [reply src/main/java/reactor/netty/udp/UdpServer.java:139]: This method is removed in master, so I'll add 1.0.0

Comment (violetagg) [reply src/main/java/reactor/netty/udp/UdpServer.java]: done

File: src/main/java/reactor/netty/udp/UdpClient.java (hunk 3 above)
Comment (violetagg) [reply src/main/java/reactor/netty/udp/UdpClient.java]: done

File: src/main/java/reactor/netty/tcp/TcpServer.java (hunk 2 above)
Comment (violetagg) [reply src/main/java/reactor/netty/tcp/TcpServer.java]: done

File: src/main/java/reactor/netty/tcp/TcpClient.java (hunk 1 above)
Comment (violetagg) [reply src/main/java/reactor/netty/tcp/TcpClient.java]: done

**Pull request merged**